vortex-buffer/src/buffer_mut.rs

```diff
@@ -322,6 +323,34 @@ impl<T> BufferMut<T> {
         self.length += slice.len();
```

Comment:
I think you should call into the unchecked function from here. This still uses checked BytesMut call